internal/wrphandlers/qos/priority_queue.go

```diff
@@ -45,20 +46,18 @@ func (pq *priorityQueue) Enqueue(msg wrp.Message) error {
 		return fmt.Errorf("%w: %v", ErrMaxMessageBytes, pq.maxMessageBytes)
 	}
 
-	// Enqueue the message and then resize until the queue no longer violates maxQueueBytes.
 	heap.Push(pq, msg)
-	for {
-		// note, total < 0 checks for an overflow
-		if !(pq.size > pq.maxQueueBytes || pq.size < 0) || pq.Len() == 0 {
-			break
-		}
-
-		// Note, `priorityQueue.drop()` does not drop the least prioritized queued message.
-		// i.e.: a high priority queued message may be drop instead of a lesser priority queued message.
+	pq.trim()
+	return nil
+}
+
+func (pq *priorityQueue) trim() {
+	// trim until the queue no longer violates maxQueueBytes.
+	for pq.sizeBytes > pq.maxQueueBytes {
+		// Note, priorityQueue.drop does not drop the least prioritized queued message.
+		// i.e.: a high priority queued message may be dropped instead of a lesser queued message.
 		pq.drop()
 	}
-
-	return nil
 }
```

Comment:
I couldn't move the trimming code to `priorityQueue.Push` without extra code with complicated swapping

Trimming after `heap.Push`'s `up(h, h.Len()-1)` makes things simpler imo 
https://github.com/golang/go/blob/4ed358b57efdad9ed710be7f4fc51495a7620ce2/src/container/heap/heap.go#L51-L54
```go
// Push pushes the element x onto the heap.
// The complexity is O(log n) where n = h.Len().
func Push(h Interface, x any) {
	h.Push(x)
	up(h, h.Len()-1)
}
```

Reply:
That is reasonable.